_overviews/toolkit/munit-resources.md

```diff
@@ -1,18 +1,86 @@
 
-See https://scalameta.org/munit/docs/fixtures.html#functional-test-local-fixtures
+## `FunFixture`
 
-Functional test-local fixture is the recommended method. Maybe mention other methods like reusable or ad-hoc fixtures 
--->
+In a test suite, you can define a `FunFixture` to initialize the resource your test needs, and to clean it up automatically after the test execution.
+
+{% tabs 'resources-1' class=tabs-scala-version %}
+{% tab 'Scala 2' %}
+```scala
+class FileTests extends munit.FunSuite:
+  val usingTempFile: FunFixture[os.Path] = FunFixture(
+    setup = _ => os.temp(prefix = "file-tests"),
+    teardown = tempFile => os.remove(tempFile)
+  )
+
+  usingTempFile.test("overwrite on file") { tempFile =>
+    os.write.over(tempFile, "Hello, World!")
+    val obtained = os.read(tempFile)
+    assertEquals(obtained, "Hello, World!")
+  }
+```
+{% endtab %}
+{% tab 'Scala 3' %}
+```scala
+class FileTests extends munit.FunSuite {
+  val usingTempFile: FunFixture[os.Path] = FunFixture(
+    setup = _ => os.temp(prefix = "file-tests"),
+    teardown = tempFile => os.remove(tempFile)
+  )
+
+  usingTempFile.test("overwrite on file") { tempFile =>
+    os.write.over(tempFile, "Hello, World!")
+    val obtained = os.read(tempFile)
+    assertEquals(obtained, "Hello, World!")
+  }
+}
+```
+{% endtab %}
+{% endtabs %}
+
+`usingTempFile` is a fixture of type `FunFixture[os.Path]`.
+It is made of two functions:
+ - The `setup` function, of type `TestOptions => os.Path`, creates a new temporary file.
+ - The `teardown` function, of type `os.Path => Unit`, deletes this temporary file.
+
+We use the `usingTempFile` fixture to define a test that needs a temporary file.
+Notice that the body of the test takes a `tempFile` as parameter.
+The fixture is responsible to create this temporary file, using its `setup` function, and to clean it up after the test, using its `teardown` function.
+It is all done automatically.
+
+In the example, we used a fixture to manage an instance of `os.Path`. In general, fixtures can manage all kind of resources, such as a table in a database, a connection to a local server, a folder...
+
+## Composing `FunFixture`s
+
+In some tests, you may need more than one resource.
+You can use `FunFixture.map2` to compose two functionnal fixtures into one.
+
+```scala
+val using2TempFiles: FunFixture[(os.Path, os.Path)] =
+  FunFixture.map2(usingTempFile, usingTempFile)
+
+using2TempFiles.test("merge tow files") {
+  case (file1, file2) =>
+    // body of the test
+}
+```
+
+Using `FunFixture.map2` on a fixture of type `FunFixture[A]` and a fixture of type `FunFixture[B]` returns a fixture of type `FunFixture[(A, B)]`.
+
+## Other fixtures
+
+`FunFixture` is the recommended type of fixture because it is the most easy to reason about: each test use its own resource in isolation.
+
+For more flexibilty, `MUnit` contains other types of fixtures: the reusable fixture, the ad-hoc fixture and the asynchronous fixture.
```

Comment:
```suggestion
For more flexibility, `MUnit` supports other types of fixtures: the reusable fixture, the ad-hoc fixture, and the asynchronous fixture.
```

To be honest “reusable fixture” does not tell much about what it does. I guess it’s a type of resource whose lifetime is scoped to an entire test suite. If this is the case, I would suggest saying it explicitly.

Reply:
There are two kinds of "reusable" fixture, the test-local one and the suite-local one. So I don't really know why they are called reusable test fixture? As if the `FunFixture` is not reusable? To me it's rather a question of Functional style versus object oriented style.

I don't want to expand more on them because IMO the `FunFixture` is the only one we need in a large majority of cases. Curious people can read the MUnit document and make their own opinion.